--- ctdb-2.0/utils/pmda/pmda_ctdb.c.orig	2012-12-22 18:36:54.933169295 +0100
+++ ctdb-2.0/utils/pmda/pmda_ctdb.c	2012-12-22 19:04:46.719801116 +0100
@@ -423,10 +423,10 @@
 		atom->ul = stats->pending_calls;
 		break;
 	case 15:
-		atom->ul = stats->lockwait_calls;
+		atom->ul = stats->locks.num_calls;
 		break;
 	case 16:
-		atom->ul = stats->pending_lockwait_calls;
+		atom->ul = stats->locks.num_pending;
 		break;
 	case 17:
 		atom->ul = stats->childwrite_calls;
@@ -450,7 +450,7 @@
 		atom->d = stats->call_latency.max;
 		break;
 	case 24:
-		atom->d = stats->lockwait_latency.max;
+		atom->d = stats->locks.latency.max;
 		break;
 	case 25:
 		atom->d = stats->childwrite_latency.max;
--- ctdb-2.0/Makefile.in.orig	2012-12-22 19:49:01.286412453 +0100
+++ ctdb-2.0/Makefile.in	2012-12-22 20:40:38.173014572 +0100
@@ -46,7 +46,7 @@
 
 PMDA_LIBS = -lpcp -lpcp_pmda
 PMDA_INSTALL = @CTDB_PMDA_INSTALL@
-PMDA_DEST_DIR = /var/lib/pcp/pmdas
+PMDA_DEST_DIR = /var/lib/pcp/pmdas/ctdb
 
 CFLAGS=@CPPFLAGS@ -g -I$(srcdir)/include -Iinclude -Ilib -Ilib/util -I$(srcdir) \
        $(TALLOC_CFLAGS) $(TEVENT_CFLAGS) $(TDB_CFLAGS) -I@libreplacedir@ \
@@ -368,10 +368,10 @@
 	if [ ! -f $(DESTDIR)$(etcdir)/ctdb/gcore_trace.sh ];then ${INSTALLCMD} -m 755 config/gcore_trace.sh $(DESTDIR)$(etcdir)/ctdb; fi
 
 install_pmda:
-	$(INSTALLCMD) -m 755 -d $(PMDA_DEST_DIR)
-	$(INSTALLCMD) -m 755 pmda/Install pmda/Remove $(PMDA_DEST_DIR)
-	$(INSTALLCMD) -m 644 pmda/pmns pmda/domain.h pmda/help pmda/README $(PMDA_DEST_DIR)
-	$(INSTALLCMD) -m 755 bin/pmdactdb $(PMDA_DEST_DIR)
+	$(INSTALLCMD) -m 755 -d $(DESTDIR)$(PMDA_DEST_DIR)
+	$(INSTALLCMD) -m 755 utils/pmda/Install utils/pmda/Remove $(DESTDIR)$(PMDA_DEST_DIR)
+	$(INSTALLCMD) -m 644 utils/pmda/pmns utils/pmda/domain.h utils/pmda/help utils/pmda/README $(DESTDIR)$(PMDA_DEST_DIR)
+	$(INSTALLCMD) -m 755 bin/pmdactdb $(DESTDIR)$(PMDA_DEST_DIR)
 
 # Should use $(datarootdir) but older autoconfs don't do this.  :-(
 install_tests: all
