--- ctdb-2.0/ib/ibwrapper.c.orig	2012-10-31 02:33:25.000000000 +0100
+++ ctdb-2.0/ib/ibwrapper.c	2012-12-16 16:56:36.987433136 +0100
@@ -34,7 +34,7 @@
 #include <unistd.h>
 
 #include "includes.h"
-#include "lib/events/events.h"
+#include "tevent.h"
 #include "ibwrapper.h"
 
 #include <infiniband/kern-abi.h>
--- ctdb-2.0/ib/ibw_ctdb_init.c.orig	2012-10-31 02:33:25.000000000 +0100
+++ ctdb-2.0/ib/ibw_ctdb_init.c	2012-12-16 17:00:33.467428207 +0100
@@ -21,6 +21,7 @@
  */
 
 #include "includes.h"
+#include "lib/util/dlinklist.h"
 #include <system/network.h>
 #include <assert.h>
 #include "ctdb_private.h"
--- ctdb-2.3/Makefile.in.orig	2013-07-11 07:18:06.000000000 +0200
+++ ctdb-2.3/Makefile.in	2013-07-13 22:20:14.932379870 +0200
@@ -293,7 +293,7 @@
 CTDB_TEST_C = $(CTDB_CLIENT_OBJ:.o=.c) tools/ctdb.c tools/ctdb_vacuum.c  tests/src/ctdb_test.c
 
 CTDB_TEST_OBJ =  $(TALLOC_OBJ) $(TDB_OBJ) \
-	@CTDB_SYSTEM_OBJ@ $(REPLACE_OBJ) $(EXTRA_OBJ) $(TEVENT_OBJ) $(SOCKET_WRAPPER_OBJ)
+	@CTDB_SYSTEM_OBJ@ $(REPLACE_OBJ) $(EXTRA_OBJ) $(TEVENT_OBJ) $(SOCKET_WRAPPER_OBJ) @INFINIBAND_WRAPPER_OBJ@
 
 tests/src/ctdb_takeover_tests.o: tests/src/ctdb_takeover_tests.c $(CTDBD_TEST_C)
 
@@ -313,9 +313,9 @@
 	@echo Linking $@
 	$(WRAPPER) $(CC) $(CFLAGS) -o $@ tests/src/ctdb_tool_stubby.o $(CTDB_TEST_OBJ) $(POPT_OBJ) $(LIB_FLAGS)
 
-tests/bin/ibwrapper_test: $(CTDB_CLIENT_OBJ) ib/ibwrapper_test.o
+tests/bin/ibwrapper_test: $(CTDB_CLIENT_OBJ) ib/ibwrapper_test.o @INFINIBAND_WRAPPER_OBJ@
 	@echo Linking $@
-	$(WRAPPER) $(CC) $(CFLAGS) -o $@ ib/ibwrapper_test.o $(CTDB_CLIENT_OBJ) $(LIB_FLAGS)
+	$(WRAPPER) $(CC) $(CFLAGS) -o $@ ib/ibwrapper_test.o $(CTDB_CLIENT_OBJ) @INFINIBAND_WRAPPER_OBJ@ $(LIB_FLAGS)
 
 manpages:
 	$(MAKE) -C doc
